### ISSUE TITLE
`isympy -c python` tab triggered auto completion not working

### ISSUE THREAD

**Issue by stevenleeS0ht** [sympy/sympy#21716]
It seems auto completion is supported in `isympy -c python` according to `https://github.com/sympy/sympy/blob/85e9c6c5882e8b628f5717770ce1f1439a204647/sympy/interactive/session.py#L266`.

However, when I typing `isympy -c python`, the `SymPyConsole` started, it when I type `si` in console, and press tab, no indent appear, so it is intend to work, but nothing out.

No matter how many times I hit tab, the cursor is still there, nothing change.

**Comment by oscarbenjamin** [comment]
Do you have ipython installed?

**Comment by stevenleeS0ht** [comment]
@oscarbenjamin , `isympy -c python` doesn't depend on `ipython`. It use `code.InteractiveConsole` and `readline` to autocomplete.

The python console have autocomplete ability.

```
Type "help", "copyright", "credits" or "license" for more information.
>>> he   
help(  hex(   
>>> st   
staticmethod(  str(           
>>> p 
pass       pow(       print(     property(  
>>>  

```

**Comment by stevenleeS0ht** [comment]
@oscarbenjamin , maybe this is relevant https://docs.python.org/3/library/rlcompleter.html

**Comment by stevenleeS0ht** [comment]
@oscarbenjamin I have done some research on this.

The possible reason maybe that https://github.com/sympy/sympy/blob/85e9c6c5882e8b628f5717770ce1f1439a204647/sympy/interactive/session.py#L266,

the readline is imported inside method, so its lifespan will be ended when __init__ method finish.

Even if put `import readline` to global scope, the problem still not solved.

regarding the following toy example:

```python3
#! /usr/bin/env python3

import code
import readline
import rlcompleter

def hi():
    print("hi")

class ToyConsole(code.InteractiveConsole):
    def __init__(self):
        code.InteractiveConsole.__init__(self)
        readline.parse_and_bind('tab: complete')
        
        
ToyConsole().interact()

```

when I run it and type the following input

```
Type "help", "copyright", "credits" or "license" for more information.
(ToyConsole)
>>> h 
hasattr(  hash(     help(     hex(      hi(       
>>> def hello():
...     print('hello')
... 
>>> h 
hasattr(  hash(     help(     hex(      hi(       
>>> he
help(  hex(   
>>>   
```

we can see, that when typing single character `h`, and then press tab button. `hi(` is prompted instead of `hello(`.

This means that readline or rlcompleter use the context of script instead of new InteractiveConsole.

The main problem now is how to make readline utilize the context of new InteractiveConsole.

@oscarbenjamin , do you have any idea?

**Comment by stevenleeS0ht** [comment]
@smichr , @asmeurer , @mattpap  any ideas?

**Comment by stevenleeS0ht** [comment]
Now, `isympy -c python` should behave properly.

**Comment by oscarbenjamin** [comment]
Should this be closed now that #21719 is merged?

**Comment by stevenleeS0ht** [comment]
@oscarbenjamin , probability yes!
